v2/manager.go

```diff
@@ -568,32 +585,38 @@ func (c *Manager) MemoryEventFD() (int, uint32, error) {
 		return 0, 0, errors.New("failed to create inotify fd")
 	}
 	wd, err := syscall.InotifyAddWatch(fd, fpath, unix.IN_MODIFY)
-	if wd < 0 {
+	if err != nil {
 		syscall.Close(fd)
-		return 0, 0, fmt.Errorf("failed to add inotify watch for %q", fpath)
+		return 0, 0, fmt.Errorf("failed to add inotify watch for %q: %w", fpath, err)
+	}
+	// monitor to detect process exit/cgroup deletion
+	evpath := filepath.Join(c.path, "cgroup.events")
+	_, err = syscall.InotifyAddWatch(fd, evpath, unix.IN_MODIFY)
```

Comment:
This means number of watches will be doubled for all existing systems, right?

Reply:
Ah, good to know the difference between instances and watches! I was thinking that there might be some correlation between watches and max pods on node configurable or something. Seems OK, feel free to mark as resolved :+1: